### ISSUE TITLE
Macro `t` with template string id could not be extracted

### ISSUE THREAD

**Issue by chuyik** [lingui/js-lingui#957]
**Describe the bug**
Macro `t` will NOT be extracted to message catalog, if we pass id with a template string like ``t({id: `id`})``.

**To Reproduce**

```jsx
import { t } from '@lingui/macro'

export default function App() {
  return (
    <div>
      <p>
        {/* id with string */}
        {t({
          id: 'today1',
          message: `Today is ${new Date()}`,
        })}
      </p>

      <p>
        {/* id with template literals */}
        {t({
          id: `today2`,
          message: `Today is ${new Date()}`,
        })}
      </p>
    </div>
  )
}

```

Only `today1` is extracted, but `today2` is missing:
```json
{
  "today1": {
    "translation": "Today is {0}",
    "message": "Today is {0}",
    "extractedComments": []
  }
}
```

**Expected behavior**
`today 2` should also be extracted.

**Additional context**

- jsLingui version `3.4.0`
- Babel version `@babel/core@7.12.10`

**Comment by dmn-bernard** [comment]
Yup this is pity I work on a project where backticks are enforced via eslint. Now I have to copy paste `// eslint-disable-line` everytime I use  the `t` macro with an id 😓

**Comment by semoal** [comment]
Just for curisosity, do you enforce the backticks always? The default rule by eslint is to use singlequote or doublequote, and if you're using + "" + "", replace it with backticks. I don't get the utility of forcing backticks to every string.

Anyways, this behaviour could be a good addition that's why triage accepted, but couldn't find time to work on it since isn't urgent at all.

**Comment by doc-sultan** [comment]
I think developers should be free to use any eslint rules on their own choice. 
I would like to try to fix that issue, but before start doing it I would like to know, is this fix not against the philosophy of lingui maintainers or maybe some other reasons were to implement it the way it is?

**Comment by semoal** [comment]
> I think developers should be free to use any eslint rules on their own choice.
> I would like to try to fix that issue, but before start doing it I would like to know, is this fix not against the philosophy of lingui maintainers or maybe some other reasons were to implement it the way it is?

The "common" behaviour is to define id like a simple string with double/single quotes, that's why the standard implementation works. When Tom refactored the v2 codebase to v3, probably didn't though about this use case, but is not against the philosophy of Lingui, it's just okay, (no free time on my side, and still less on Tom side)

Feel free to take it and ping me if you need some guidance =)

**Comment by semoal** [comment]
Released this feature on 3.8.3 guys!
Thanks to everyone and special to thanks to @sultan99 who resolved this issue!
